pkg/planner/core/operator/logicalop/logical_plans_misc.go

```diff
@@ -88,20 +88,18 @@ func addSelection(p base.LogicalPlan, child base.LogicalPlan, conditions []expre
 		p.Children()[chIdx] = child
 		return
 	}
-	conditions = expression.PropagateConstant(p.SCtx().GetExprCtx(), conditions)
+	conditions = utilfuncp.ApplyPredicateSimplification(p.SCtx(), conditions)
+	if len(conditions) == 0 {
```

Comment:
ApplyPredicateSimplification is a super set of PropagateConstant？

Reply:
Yes

Reply:
Updated. 

I find that although ```ApplyPredicateSimplification``` will call ```PropagateConstant```. But it adds too many conditions to avoid calling. I have found the test case which proves this point https://github.com/pingcap/tidb/pull/61571/files